app/waf/src/controllers/declaration.controller.ts

```diff
@@ -172,6 +175,29 @@ export class DeclarationController extends BaseController {
         defs: defs,
       };
     }
+
+    if (service.iRules && service.iRules.length > 0) {
+      let refs: (string | object)[] = [];
+      let defs: {[k: string]: object} = {};
+
+      for (let i of service.iRules) {
```

Comment:
Maybe we can add a 'findByIds()' in repository class to load all iRules in one db call. Let's do it later.

Reply:
I need to check the iRule ID one by one so that to determine using 'bigip' or 'use'. 
Can `findByIds` work when iRule with not-exist-ID?

Reply:
I think it should return a set of result, which includes the iRules in db. Then we can go through the ids, if it is not in the result set, it can use 'bigip' to load iRule from '''/Common''' partition, right? Let's think that later.